phoenix-core/src/main/java/org/apache/phoenix/compile/ProjectionCompiler.java

```diff
@@ -507,6 +507,9 @@ public static RowProjector compile(StatementContext context, SelectStatement sta
                 } else {
                     for (byte[] cq : entry.getValue()) {
                             PColumn column = family.getPColumnForColumnQualifier(cq);
+                            // Continue: If an EMPTY_COLUMN is in the projection list, since
+                            // no new KeyValue is returned.
+                            if (column == null) continue;
```

Comment:
Don't we omit the empty column when projecting? Under which circumstances will this condition be true?

Reply:
This check is for the case when we add empty column in the projection list and since empty column is not part of the table column list, we need to bypass further processing.